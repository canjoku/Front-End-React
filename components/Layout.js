import { Component } from 'react'
import Head from 'next/head'


import Header from '../components/Header'
import Footer from '../components/Footer'
import '../scss/application.scss'



class Layout extends Component {
  render () {
    const { mutation } = this.props
    return (
      <div>
        <Head>
          <title>Chidi Njoku</title>
          <link rel='stylesheet' href='/_next/static/style.css' />
          <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
        </Head>
        <Header mutation={mutation}/>
          { this.props.children }
        <Footer />
      </div>
    )
  }
}

export default Layout
