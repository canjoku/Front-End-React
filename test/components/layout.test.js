import Layout from '../../components/Layout'

test('<Layout /> renders correctly', () => {
  const component = shallow(<Layout />)
  expect(component).toMatchSnapshot()
})
