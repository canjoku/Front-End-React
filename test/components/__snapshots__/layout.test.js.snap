// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SideEffect(Head)>
          <title>
            Chidi Njoku
          </title>
          <link
            href="/_next/static/style.css"
            rel="stylesheet"
          />
          <meta
            content="width=device-width, initial-scale=1, maximum-scale=1.0"
            name="viewport"
          />
        </SideEffect(Head)>,
        <Header
          mutation={undefined}
        />,
        undefined,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <title>
              Chidi Njoku
            </title>,
            <link
              href="/_next/static/style.css"
              rel="stylesheet"
            />,
            <meta
              content="width=device-width, initial-scale=1, maximum-scale=1.0"
              name="viewport"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Chidi Njoku",
            },
            "ref": null,
            "rendered": "Chidi Njoku",
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "/_next/static/style.css",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "width=device-width, initial-scale=1, maximum-scale=1.0",
              "name": "viewport",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "mutation": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideEffect(Head)>
            <title>
              Chidi Njoku
            </title>
            <link
              href="/_next/static/style.css"
              rel="stylesheet"
            />
            <meta
              content="width=device-width, initial-scale=1, maximum-scale=1.0"
              name="viewport"
            />
          </SideEffect(Head)>,
          <Header
            mutation={undefined}
          />,
          undefined,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <title>
                Chidi Njoku
              </title>,
              <link
                href="/_next/static/style.css"
                rel="stylesheet"
              />,
              <meta
                content="width=device-width, initial-scale=1, maximum-scale=1.0"
                name="viewport"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Chidi Njoku",
              },
              "ref": null,
              "rendered": "Chidi Njoku",
              "type": "title",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "/_next/static/style.css",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "width=device-width, initial-scale=1, maximum-scale=1.0",
                "name": "viewport",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "mutation": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
